### PR TITLE
Add hiring link

### PR DESCRIPTION
**Pull request by anthonydillon**
## Done
Added a hiring link option to the list of link in the global nav.
Added README usage docs
Bump the version to 2.3.0

## QA
- Run `./run`
- Open http://172.17.0.4:8300/
- Check there is no difference and no hiring link
- Open the index.html file
- Update the `createNav` function with the following:
``` html
<script>
  canonicalGlobalNav.createNav({
    hiring: 'https://boards.greenhouse.io/canonical/jobs/1586585',
  });
</script>
```
- Reload the page and check you see the "We are hiring" link on the footer
- Check the hiring value to something other then a link
- Reload and check it doesn't render the link

## Screenshot
Screenshot_2019-11-14 Canonical global navigation demo

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,12 +57,22 @@ For example, to set the `max-width` to `80rem`:
 </script>
 ```
 
-If you're importing;
+- `hiring`: A link to a hiring page which will display a link in the header to the job page (default: `false`)
```
Comment (sparkiegeek) [inline README.md]: s/A link to/A URL for/ ?

File: README.md [hunk 2]
```diff
@@ -57,12 +57,22 @@ For example, to set the `max-width` to `80rem`:
...
 
-```js
-import { createNav } from '@canonical/global-nav';
+For example, to set the `hiring` to `https://boards.greenhouse.io/canonical/jobs/1586585`
+
+````html
```
Comment (huwshimi) [inline README.md]: This has an extra `

File: README.md (hunk 1 above)
Comment (anthonydillon) [reply README.md]: Done thanks

File: README.md [hunk 3]
```diff
@@ -57,12 +57,22 @@ For example, to set the `max-width` to `80rem`:
...
+<script src="/node_modules/@canonical/global-nav/dist/index.js"></script>
+
+<script>
+  canonicalGlobalNav.createNav({
+    hiring: 'https://boards.greenhouse.io/canonical/jobs/1586585',
+  });
+</script>
+
+If you're importing; ```js import { createNav } from '@canonical/global-nav';
```
Comment (huwshimi) [inline README.md]: I think there should be a code end block "```"  above this line and the new block that starts on this line should be moved to the next line.

File: README.md [hunk 4]
```diff
@@ -57,12 +57,22 @@ For example, to set the `max-width` to `80rem`:
...
 createNav({ maxWidth: '80rem' });
-```
+````
```
Comment (huwshimi) [inline README.md]: The should get moved above "If you're importing..."

File: src/js/global-nav.js [hunk 5]
```diff
@@ -274,7 +276,30 @@ function addListeners(breakpoint, wrapper) {
   overlay.addEventListener('click', closeNav);
 }
 
-export const createNav = ({ maxWidth = '68rem' } = {}) => {
+function validURL(str) {
+  const pattern = new RegExp(
+    '^(https?:\\/\\/)?' + // protocol
```
Comment (huwshimi) [inline src/js/global-nav.js]: I think this has to start with http(s):// (or else append it) otherise you can have links like `hiring: "example.com"` which would result in "http://dev.local:8300/example.com".

File: README.md (hunk 2 above)
Comment (anthonydillon) [reply README.md]: done

File: README.md (hunk 3 above)
Comment (anthonydillon) [reply README.md]: done

File: README.md (hunk 4 above)
Comment (anthonydillon) [reply README.md]: done

File: src/js/global-nav.js (hunk 5 above)
Comment (anthonydillon) [reply src/js/global-nav.js]: Removed the validation as there might be a few ways we want to link like interal anchors so safer without it.

**Comment by anthonydillon** [comment]
@huwshimi thanks, addressed your points.

**Comment by huwshimi** [comment]
@anthonydillon looks good now!

**Pull request merged**